src/katsdpcontroller/product_controller.py

```diff
@@ -517,7 +517,7 @@ def __init__(
         )
         self._state_sensor = Sensor(
             ProductState,
-            "state",
+            "product-state",
```

Comment:
The internal object is conveniently also called `ProductState`, which confirms that the new name is a good one :-)